README.md

```diff
@@ -1,28 +1,60 @@
+[![Documentation Status](https://readthedocs.org/projects/numpyro/badge/?version=latest)](https://numpyro.readthedocs.io/en/latest/?badge=latest)
 
-[Pyro](https://github.com/pyro-ppl/pyro) on Numpy. This uses 
-[JAX](https://github.com/google/jax) for autograd and JIT support. This is an 
-early stage experimental library that is under active development, and there are 
-likely to be  many changes to the API and internal classes, as the design evolves. 
+Probabilistic programming with Numpy on GPUs powered by [JAX](https://github.com/google/jax) for autograd and JIT support. 
 
- ## Design Goals
+## What is NumPyro?
  
- - **Lightweight** - We do not intend to reimplement any heavy inference machinery 
-   from Pyro, but would like to provide a flexible substrate that can be built 
-   upon. We will provide support for Pyro primitives like `sample` and `param` 
-   which can be interpreted with side-effects using effect handlers. Users should 
-   be able to extend this to implement custom inference algorithms, and write 
-   their models using the familiar Numpy API.
- - **Functional** - The API for the inference algorithms and other utility functions 
-   may deviate from Pyro in favor of a more *functional* style that works better 
-   with JAX. e.g. no global param store or random state.
- - **Fast** - Using JAX, we aim to aggressively JIT compile intermediate computations 
-   to XLA optimized kernels. We will evaluate JIT compilation, and benchmark runtime 
-   for Hamiltonian Monte Carlo.
-    
- ## Longer-term Plans
+NumPyro is a small probabilistic programming library built on [JAX](https://github.com/google/jax). It essentially provides a NumPy backend for [Pyro](https://github.com/pyro-ppl/pyro), with some minor changes to the inference API and syntax. Since we use JAX, we get autograd and JIT compilation to GPUs for free. This is an alpha release, and the API is subject to change as the design evolves.
+ 
+NumPyro is designed to be *lightweight* and focuses on providing a flexible substrate that users can build on:
+
+ - **Pyro Primitives:** NumPyro programs can contain regular Python and NumPy code, in addition to [Pyro primitives](http://pyro.ai/examples/intro_part_i.html) like `sample` and `param`. The model code should look very similar to Pyro except for some minor differences between PyTorch and Numpy's API. See [Examples](#Examples).
+ - **Inference algorithms:** NumPyro currently supports Hamiltonian Monte Carlo, including an implementation of the No U-Turn Sampler. One of the motivations for NumPyro was to speed up Hamiltonian Monte Carlo by JIT compiling the verlet integration step that includes multiple gradient computations. With JAX, we can compose `jit` and `grad` to compile the entire integration step into an XLA optimized kernel. We also eliminate Python overhead by JIT compiling the entire tree building stage in NUTS. There is also a basic Variational Inference implementation for reparameterized distributions.
```

Comment:
I would like to add a link to iterative NUTS at the end of "We also eliminate Python overhead by JIT compiling the entire tree building stage in NUTS (this is possible using [Iterative NUTS](link))". What do you think?

Reply:
Sure, will add the link to the wiki. I think it will be best to publish that into a separate technical report / whitepaper, along with benchmarks soon!